editor/plugins/script_text_editor.cpp

```diff
@@ -1731,65 +1725,108 @@ ScriptTextEditor::ScriptTextEditor() {
-
 	goto_menu->get_popup()->add_shortcut(ED_GET_SHORTCUT("script_text_editor/goto_function"), SEARCH_LOCATE_FUNCTION);
 	goto_menu->get_popup()->add_shortcut(ED_GET_SHORTCUT("script_text_editor/goto_line"), SEARCH_GOTO_LINE);
 	goto_menu->get_popup()->add_separator();
 
-	bookmarks_menu = memnew(PopupMenu);
-	bookmarks_menu->set_name("Bookmarks");
 	goto_menu->get_popup()->add_child(bookmarks_menu);
 	goto_menu->get_popup()->add_submenu_item(TTR("Bookmarks"), "Bookmarks");
 	_update_bookmark_list();
 	bookmarks_menu->connect("about_to_popup", callable_mp(this, &ScriptTextEditor::_update_bookmark_list));
 	bookmarks_menu->connect("index_pressed", callable_mp(this, &ScriptTextEditor::_bookmark_item_pressed));
 
-	breakpoints_menu = memnew(PopupMenu);
-	breakpoints_menu->set_name("Breakpoints");
 	goto_menu->get_popup()->add_child(breakpoints_menu);
 	goto_menu->get_popup()->add_submenu_item(TTR("Breakpoints"), "Breakpoints");
 	_update_breakpoint_list();
 	breakpoints_menu->connect("about_to_popup", callable_mp(this, &ScriptTextEditor::_update_breakpoint_list));
 	breakpoints_menu->connect("index_pressed", callable_mp(this, &ScriptTextEditor::_breakpoint_item_pressed));
 
-	quick_open = memnew(ScriptEditorQuickOpen);
-	add_child(quick_open);
-	quick_open->connect("goto_line", callable_mp(this, &ScriptTextEditor::_goto_line));
+	goto_menu->get_popup()->connect("id_pressed", callable_mp(this, &ScriptTextEditor::_edit_option));
+}
 
-	goto_line_dialog = memnew(GotoLineDialog);
-	add_child(goto_line_dialog);
+ScriptTextEditor::ScriptTextEditor() {
+	code_editor = memnew(CodeTextEditor);
+	code_editor->add_theme_constant_override("separation", 2);
+	code_editor->set_anchors_and_margins_preset(Control::PRESET_WIDE);
+	code_editor->set_code_complete_func(_code_complete_scripts, this);
+	code_editor->set_v_size_flags(SIZE_EXPAND_FILL);
+
+	warnings_panel = memnew(RichTextLabel);
+	warnings_panel->set_custom_minimum_size(Size2(0, 100 * EDSCALE));
+	warnings_panel->set_h_size_flags(SIZE_EXPAND_FILL);
+	warnings_panel->set_meta_underline(true);
+	warnings_panel->set_selection_enabled(true);
+	warnings_panel->set_focus_mode(FOCUS_CLICK);
+	warnings_panel->hide();
+
+	update_settings();
+
+	code_editor->get_text_edit()->set_callhint_settings(
+			EditorSettings::get_singleton()->get("text_editor/completion/put_callhint_tooltip_below_current_line"),
+			EditorSettings::get_singleton()->get("text_editor/completion/callhint_tooltip_offset"));
+
+	code_editor->get_text_edit()->set_select_identifiers_on_hover(true);
+	code_editor->get_text_edit()->set_context_menu_enabled(false);
+
+	context_menu = memnew(PopupMenu);
+
+	color_panel = memnew(PopupPanel);
+
+	edit_hb = memnew(HBoxContainer);
+
+	edit_menu = memnew(MenuButton);
+	edit_menu->set_text(TTR("Edit"));
+	edit_menu->set_switch_on_hover(true);
+
+	convert_case = memnew(PopupMenu);
+	convert_case->set_name("convert_case");
+
+	highlighters[TTR("Standard")] = nullptr;
```

Comment:
CI complains here: 

```
editor/plugins/script_text_editor.cpp: In constructor 'ScriptTextEditor::ScriptTextEditor()':
editor/plugins/script_text_editor.cpp:1788:34: error: ambiguous overload for 'operator=' (operand types are 'Ref<EditorSyntaxHighlighter>' and 'std::nullptr_t')
 1788 |  highlighters[TTR("Standard")] = nullptr;
      |                                  ^~~~~~~
```

```suggestion
	highlighters[TTR("Standard")] = Ref<EditorSyntaxHighlighter>();
```

Reply:
The line can be removed, the "Standard" one is no longer `null` and they are both already generated and added [here.](https://github.com/godotengine/godot/blob/ca1a5bdcbe28c38b745a0c0cbdbbe42708069e35/editor/plugins/script_text_editor.cpp#L1719-L1726)